static-assets/components/cstudio-admin/mods/content-type-propsheet/template.js

```diff
@@ -91,41 +91,88 @@ YAHOO.extend(
 
         this.controlsContainerEl = controlsContainerEl;
 
-        editEl.onclick = function() {
-          var contentType = _self.valueEl.value;
+        editEl.onclick = function () {
+          const path = _self.valueEl.value;
+          const contentType = _self.currentContenType.contentType;
+          const onCreatedId = 'createFileDialogOnCreated';
+          const onCloseId = 'createFileDialogOnClosed';
+          let unsubscribe, cancelUnsubscribe;
+
+          const showCodeEditor = (path, contentType) => {
+            CrafterCMSNext.system.store.dispatch({
+              type: 'SHOW_CODE_EDITOR_DIALOG',
+              payload: {
+                path,
+                contentType,
+                mode: 'ftl'
+              }
+            });
+          };
+
+          if (path === '') {
+            CrafterCMSNext.system.store.dispatch({
+              type: 'SHOW_CREATE_FILE_DIALOG',
+              payload: {
+                path: '/templates/web',
+                type: 'template',
+                onCreated: {
+                  type: 'BATCH_ACTIONS',
+                  payload: [
+                    {
+                      type: 'DISPATCH_DOM_EVENT',
+                      payload: { id: onCreatedId }
+                    },
+                    { type: 'CLOSE_CREATE_FILE_DIALOG' }
+                  ]
+                },
+                onClosed: {
+                  type: 'BATCH_ACTIONS',
+                  payload: [
+                    {
+                      type: 'DISPATCH_DOM_EVENT',
+                      payload: { id: onCloseId }
+                    },
+                    { type: 'CREATE_FILE_DIALOG_CLOSED' }
+                  ]
+                }
+              }
+            });
 
-          if (contentType == '') {
-            CStudioAuthoring.Operations.createNewTemplate(null, function(templatePath) {
+            unsubscribe = CrafterCMSNext.createLegacyCallbackListener(onCreatedId, (response) => {
```

Comment:
Remember we don't need two of these. Just one plus sending an additional type argument on the dispatch payload suffices